File: 2026POTD/JANUARY/06-01-2026/1161.kt
class Solution {
    fun maxLevelSum(root: TreeNode?): Int {
        val q: ArrayDeque<TreeNode> = ArrayDeque()
        q.add(root!!)
        var level = 1
        var ans = 1
        var maxSum = Long.MIN_VALUE

        while (q.isNotEmpty()) {
            val size = q.size
            var sum = 0L

            repeat(size) {
                val node = q.removeFirst()
                sum += node.`val`
                node.left?.let { q.add(it) }
                node.right?.let { q.add(it) }
            }

            if (sum > maxSum) {
                maxSum = sum
                ans = level
            }
            level++
        }
        return ans
    }
}

File: 2026POTD/JANUARY/06-01-2026/1161.dart
class Solution {
  int maxLevelSum(TreeNode? root) {
    List<TreeNode> q = [root!];
    int level = 1, ans = 1;
    int maxSum = -1 << 60;

    while (q.isNotEmpty) {
      int size = q.length;
      int sum = 0;

      for (int i = 0; i < size; i++) {
        TreeNode node = q.removeAt(0);
        sum += node.val;
        if (node.left != null) q.add(node.left!);
        if (node.right != null) q.add(node.right!);
      }

      if (sum > maxSum) {
        maxSum = sum;
        ans = level;
      }
      level++;
    }
    return ans;
  }
}